holidays/countries/germany.py

```diff
@@ -168,59 +178,92 @@ def _populate_subdiv_by_public_holidays(self):
         self._add_all_saints_day(tr("Allerheiligen"))
 
     def _populate_subdiv_byp_public_holidays(self):
+        if self._year <= 1989:
+            return None
+
         if self._year >= 1991:
             self._add_epiphany_day(tr("Heilige Drei Könige"))
             self._add_corpus_christi_day(tr("Fronleichnam"))
 
         self._add_all_saints_day(tr("Allerheiligen"))
 
     def _populate_subdiv_hb_public_holidays(self):
+        if self._year <= 1989:
+            return None
+
```

Comment:
👍

Reply:
I can't wait for class level years check migration. This is too repetitive.